### PR TITLE
Bugfix deviation replace/add config should check the target node's config tree

### LINKED ISSUE #219: deviation replace config isn't working for top level nodes
``` YANG
module d {
  namespace urn:d;
  prefix d;

  container x {
    choice c {
      leaf d {
        type string;
      }
    }
  }

  deviation /x/c/d {
    deviate not-supported;
  }

  deviation /x {
    deviate replace {
        config false;
    }
  }
}
```

However RFC 6020 Section [7.19.1 ](https://tools.ietf.org/html/rfc6020#section-7.19.1) suggested

`If a node has "config" set to "false", no node underneath 
it can have "config" set to "true".`

When i get tree output with `pyang` the `choice c` setted as `config true`

``` SHELL
$ pyang -f tree deviations/d.yang
module: d
   +--ro x
      +--rw (c)?
```

### PR DESCRIPTION
**Pull request by fredgan**
Hi, This PR fix #219 

Meanwhile, another issue is found and fixed: 

When a comment substatement is in the deviate statement, it will report an error like
```
the "_comment" property does not exist in node "main::x"
```
However, I don't think this is an error.

### CODE REVIEW

**Comment by fredgan** [comment]
I find another issue to be solved, just wait.

File: pyang/statements.py [hunk 1]
```diff
@@ -2193,45 +2245,42 @@ def v_reference_deviate(ctx, stmt):
...
-                            # maybe fix the statment
-                            old = d.search_one('config')
-                            if old is None:
-                                negc = copy.copy(c)
-                                if c.arg == 'true':
-                                    negc.arg = 'false'
-                                else:
-                                    negc.arg = 'true'
-                                d.substmts.append(negc)
+                    err_add(ctx.errors, t.pos, 'BAD_DEVIATE_REP', (c.keyword, t.i_module.arg, t.arg))
```
Comment (mbj4668) [inline pyang/statements.py]: Please have a look at CONTRIBUTING; there are just two rules ;-)

File: test/test_issues/test_i219/expect/mod1.expect [hunk 2]
```diff
@@ -0,0 +1,3 @@
+mod1.yang:9: error: INVALID_CONFIG
```
Comment (mbj4668) [inline test/test_issues/test_i219/expect/mod1.expect]: Perhaps this is not due to your code, but this error is a bit weird - it refers to line 9, but line 9 is fine, the problem is on line 38.  Somehow we need to update the .pos with the deviation statement's .pos.

File: pyang/statements.py (hunk 1 above)
Comment (fredgan) [reply pyang/statements.py]: It's my fault. Didn't notice it's more than 80 characters.

File: test/test_issues/test_i219/expect/mod1.expect (hunk 2 above)
Comment (fredgan) [reply test/test_issues/test_i219/expect/mod1.expect]: OK, I will replace the .pos with the deviation statement's .pos.

File: pyang/statements.py [hunk 3]
```diff
@@ -2118,6 +2118,47 @@ def v_reference_deviation(ctx, stmt):
...
+            for children in node.i_children:
+                if search_children_config_true(children, target):
+                    return True
+            return False
+
+    def inherit_parent_i_config(node, value):
+        if node is None or not hasattr(node, 'i_children'):
+            return
+        else:
+            for children in node.i_children:
```
Comment (mbj4668) [inline pyang/statements.py]: Minor, but change "children" to "child".

File: pyang/statements.py [hunk 4]
```diff
@@ -2118,6 +2118,47 @@ def v_reference_deviation(ctx, stmt):
     stmt.i_target_node = find_target_node(ctx, stmt)
 
 def v_reference_deviate(ctx, stmt):
+    def search_ancestor_config_false(node, target):
```
Comment (mbj4668) [inline pyang/statements.py]: I wonder if this is correct / necessary.  I think it should be sufficient to check parent.i_config.  Did you consider that?

File: pyang/statements.py (hunk 3 above)
Comment (fredgan) [reply pyang/statements.py]: OK

File: pyang/statements.py (hunk 4 above)
Comment (fredgan) [reply pyang/statements.py]: Yes, It's a good point. I will try it tomorrow.

**Pull request merged**